### PR TITLE
feat: make unprotecting a note a protected action

### PR DESCRIPTION
**Pull request by antsgar**
Use new SNJS protectNote and unprotectNote methods where unprotecting a note is now a protected action.

### CODE REVIEW

File: src/screens/SideMenu/NoteSideMenu.tsx [hunk 1]
```diff
@@ -418,44 +418,62 @@ export const NoteSideMenu = React.memo((props: Props) => {
...
-            }
-          );
-        }
-      }
-      if (!application?.findItem(note.uuid)) {
-        application?.alertService!.alert(
-          "The note you are attempting to save can not be found or has been deleted. Changes you make will not be synced. Please copy this note's text and start a new note."
+    if (note.deleted) {
+      application?.alertService?.alert(
+        'The note you are attempting to edit has been deleted, and is awaiting sync. Changes you make will be disregarded.'
```
Comment (mobitar) [inline src/screens/SideMenu/NoteSideMenu.tsx:428]: Wow, didn't realize there's so much duplication between NoteSideMenu and NoteCell. Although not a blocker for this release, this is really bad design. Seems to be a pre-existing issue, but this is definitely something that needs to be refactored.

**Review by mobitar — APPROVED** [review]

File: src/screens/SideMenu/NoteSideMenu.tsx (hunk 1 above)
Comment (antsgar) [reply src/screens/SideMenu/NoteSideMenu.tsx:428]: Yeah I agree, there's a chunk of code that's literally almost the same for both components. Can probably be extracted to a hook or a util, I'll look into it.

File: src/lib/snjs_helper_hooks.ts [hunk 2]
```diff
@@ -434,3 +435,93 @@ export const useProtectNoteAlert = () => {
 
   return [showProtectNoteAlert];
 };
+
+export const useChangeNoteChecks = (
+  note: SNNote | undefined,
+  editor: Editor | undefined = undefined
+) => {
+  // Context
```
Comment (mobitar) [inline src/lib/snjs_helper_hooks.ts:443]: These comments don't seem necessary..

**Review by mobitar — APPROVED** [review]

File: src/lib/snjs_helper_hooks.ts (hunk 2 above)
Comment (antsgar) [reply src/lib/snjs_helper_hooks.ts:443]: Probably aren't, I just followed the pattern from the other hooks here, but they don't add much. I could delete them from all hooks for consistency.

Comment (antsgar) [reply src/lib/snjs_helper_hooks.ts:443]: Checked and it seems they're not just in this file but everywhere in the app, will leave them for now

**Pull request merged**